pkg/git/libgit2/transport.go

```diff
@@ -195,13 +198,28 @@ func parseKnownHosts(s string) ([]knownKey, error) {
 	return knownHosts, nil
 }
 
-func (k knownKey) matches(host string, key []byte) bool {
+func (k knownKey) matches(host string, hostkey git2go.HostkeyCertificate) bool {
 	if !containsHost(k.hosts, host) {
 		return false
 	}
 
-	hash := sha1.Sum(k.key.Marshal())
-	if bytes.Compare(hash[:], key) != 0 {
+	var fingerprint []byte
+	var hasher hash.Hash
+	switch hostkey.Kind {
```

Comment:
hostkey.Kind is a bitset, so you want something like

```go
switch {
    case hostkey.Kind & git2go.HostkeySHA256 > 0:
        ...
}
```

i.e., check the bitset against each enum value, starting with the most preferred (SHA256) down to the lowly MD5.

(if you print `hostkey.Kind` out, it'll usually have the value `7`, which is `HostkeyMD5 | HostkeySHA1 | HostkeySHA256` (the enum values are in https://github.com/libgit2/libgit2/blob/27e34f9b9843f7bcc33a4ccfe3e395fe303cba63/include/git2/cert.h#L76)